Applications/Spire/Source/Ui/TextBox.cpp

```diff
@@ -1,11 +1,69 @@
 #include "Spire/Ui/TextBox.hpp"
 #include <QEvent>
 #include <QKeyEvent>
+#include <QTimer>
 #include "Spire/Spire/Dimensions.hpp"
 
 using namespace boost::signals2;
 using namespace Spire;
 
+namespace {
+  const auto BACKGROUND_COLOR = QColor("#FFFFFF");
+  const auto BACKGROUND_WARNING_COLOR = QColor("#FFF1F1");
+  const auto BORDER_COLOR = QColor("#C8C8C8");
+  const auto BORDER_WARNING_COLOR = QColor("#B71C1C");
+  const auto WARNING_FADE_OUT_TIME_LINE_FRAME = 10;
+  const auto WARNING_FADE_OUT_TIME_MS = 300;
+  const auto WARNING_SHOW_DELAY_MS = 250;
+  const auto WARNING_PROPERTY_NAME = "warning";
+
+  const auto WARNING_STYLE_SHEET(const QColor& background_color,
```

Comment:
I think some of these methods should have lowercase names because they're not constants.

I also don't think values returned by value should be const. We typically only return references as constants because a caller could potentially modify the returned value.